src/ORM/ArrayList.php

```diff
@@ -707,11 +708,21 @@ protected function filterOrExclude(array $filters, bool $inclusive = true, bool
     {
         $itemsToKeep = [];
         $searchFilters = [];
+        $hasNullFilter = false;
 
         foreach ($filters as $filterKey => $filterValue) {
-            // Convert null to an empty string for backwards compatability, since nulls are treated specially
+            // Check if we have any null filter values for backwards compatability, since nulls are treated specially
             // in the ExactMatchFilter
-            $searchFilter = $this->createSearchFilter($filterKey, $filterValue ?? '');
+            if (is_array($filterValue)) {
+                foreach ($filterValue as $value) {
+                    if ($value === null) {
+                        $hasNullFilter = true;
+                    }
+                }
+            } elseif ($filterValue === null) {
+                $hasNullFilter = true;
+            }
+            $searchFilter = $this->createSearchFilter($filterKey, $filterValue);
```

Comment:
Don't cast the null value to empty string anymore since `0 == ''` evaluates false. Instead, just take note of if we have any null values for later.